spring-cloud-gcp-pubsub-stream-binder/src/test/java/com/google/cloud/spring/stream/binder/pubsub/config/PubSubBinderConfigurationTests.java

```diff
@@ -0,0 +1,131 @@
+import com.google.cloud.spring.pubsub.core.PubSubTemplate;
+import com.google.cloud.spring.stream.binder.pubsub.provisioning.PubSubChannelProvisioner;
+import org.junit.Test;
+
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Bean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Tests for PubSubBinderConfiguration and its interaction with other autoconfiguration classes.
+ */
+public class PubSubBinderConfigurationTests {
+
+	@Test
+	public void testEverythingEnabled_implicit() {
+		ApplicationContextRunner baseContext = new ApplicationContextRunner()
+				// no property values specified
+				.withConfiguration(AutoConfigurations.of(
+						GcpContextAutoConfiguration.class,
+						GcpPubSubAutoConfiguration.class,
+						PubSubBinderConfiguration.class))
+				.withUserConfiguration(TestConfiguration.class);
+		baseContext.run(ctx -> assertThat(ctx).hasSingleBean(PubSubChannelProvisioner.class));
+	}
+
+	@Test
+	public void testEverythingEnabled_explicit() {
+		ApplicationContextRunner baseContext = new ApplicationContextRunner()
+				.withPropertyValues(
+						// including 'core' for completeness, but it's not super relevant to the test.
+						"spring.cloud.gcp.core.enabled=true",
+						"spring.cloud.gcp.pubsub.enabled=true",
+						"spring.cloud.gcp.pubsub.binder.enabled=true")
+				.withConfiguration(AutoConfigurations.of(
+						GcpContextAutoConfiguration.class,
+						GcpPubSubAutoConfiguration.class,
+						PubSubBinderConfiguration.class))
+				.withUserConfiguration(TestConfiguration.class);
+		baseContext.run(ctx -> assertThat(ctx).hasSingleBean(PubSubChannelProvisioner.class));
+	}
+
+	@Test
+	public void testBinderDisabled() {
+		ApplicationContextRunner baseContext = new ApplicationContextRunner()
+				.withPropertyValues(
+						"spring.cloud.gcp.pubsub.enabled=true",
+						"spring.cloud.gcp.pubsub.binder.enabled=false")
+				.withConfiguration(AutoConfigurations.of(
+						GcpContextAutoConfiguration.class,
+						GcpPubSubAutoConfiguration.class,
+						PubSubBinderConfiguration.class))
+				.withUserConfiguration(TestConfiguration.class);
+		baseContext.run(ctx -> assertThat(ctx).doesNotHaveBean(PubSubChannelProvisioner.class));
+	}
+
+	@Test
+	public void testPubSubDisabled_noReplacements() {
+		ApplicationContextRunner baseContext = new ApplicationContextRunner()
+				.withPropertyValues(
+						"spring.cloud.gcp.pubsub.enabled=false",
+						"spring.cloud.gcp.pubsub.binder.enabled=true")
+				.withConfiguration(AutoConfigurations.of(
+						GcpContextAutoConfiguration.class,
+						GcpPubSubAutoConfiguration.class,
+						PubSubBinderConfiguration.class))
+				.withUserConfiguration(TestConfiguration.class);
+		baseContext.run(ctx -> assertThat(ctx).doesNotHaveBean(PubSubChannelProvisioner.class));
```

Comment:
Are there any warnings logged from Spring Cloud Stream perhaps?

Reply:
```
A component required a bean of type 'org.springframework.cloud.stream.binder.Binder' that could not be found.
```

Obviously if there are multiple binders this wouldn't happen, but at least there is some check that if this is the only one there has been a misconfiguration somewhere.